### ISSUE TITLE
[BUG] PDF-Attachment from multipart MIME encoded mail is skipped (content disposition <EMPTY>)

### ISSUE THREAD

**Issue by trurli** [paperless-ngx/paperless-ngx#6827]
### Description

When fetching emails with attachment I expected paperless to extract the PDF attachments and import them. Instead mail.log has the following debug logs for all matching emails. Emails are all from the same sender and are multipart messages with no inline attachments.

```
[2024-05-25 05:50:10,004] [DEBUG] [paperless_mail] Rule <REDACTED>: Processing mail <REDACTED> from <REDACTED> with 1 attachment(s)
[2024-05-25 05:50:10,009] [DEBUG] [paperless_mail] Rule <REDACTED>: Skipping attachment <REDACTED> with content disposition
```

Content type header from the mail is as follows:

```
Content-Type: multipart/mixed; boundary=--boundary_160_efebf789-ff64-4a9e-80e4-e81d09b55916
```

Attachment in question in the mail begins like this in the mail source:

```
----boundary_160_efebf789-ff64-4a9e-80e4-e81d09b55916
Content-Type: application/octet-stream; name=<REDACTED>.pdf
Content-Transfer-Encoding: base64

<base64 encoded PDF here>
----boundary_160_efebf789-ff64-4a9e-80e4-e81d09b55916--
```

### Steps to reproduce

1. Set up an email account
2. Set up rule to process attachments 
    Attachment Type: Process all files, including 'inline' attachments
    Consumption Scope: Only process attachments
    Filter attachment filename includes: *.pdf
3. Wait until the CRON job fetches the emails

### Webserver logs

```bash
webserver-1  | [2024-05-25 05:50:08,131] [INFO] [celery.beat] Scheduler: Sending due task Check all e-mail accounts (paperless_mail.tasks.process_mail_accounts)
webserver-1  | [2024-05-25 05:50:08,137] [INFO] [celery.worker.strategy] Task paperless_mail.tasks.process_mail_accounts[8691861a-9b9b-4281-bef2-185035ae3aae] received
webserver-1  | [2024-05-25 05:50:10,920] [INFO] [celery.app.trace] Task paperless_mail.tasks.process_mail_accounts[8691861a-9b9b-4281-bef2-185035ae3aae] succeeded in 2.77730014489498s: 'No new documents were added.'
```

### Paperless-ngx version

2.8.6

### Host OS

Arch Linux ARM (aarch64)

### Installation method

Docker - official image

### System status

```json
{"pngx_version":"2.8.6","server_os":"Linux-6.8.9-1-aarch64-ARCH-aarch64-with-glibc2.36","install_type":"docker","storage":{"total":1500279713792,"available":898878894080},"database":{"type":"postgresql","url":"paperless","status":"OK","error":null,"migration_status":{"latest_migration":"socialaccount.0006_alter_socialaccount_extra_data","unapplied_migrations":[]}},"tasks":{"redis_url":"redis://broker:6379","redis_status":"OK","redis_error":null,"celery_status":"OK","index_status":"OK","index_last_modified":"2024-05-25T05:32:07.430897Z","index_error":null,"classifier_status":"OK","classifier_last_trained":"2024-05-25T07:05:01.050031Z","classifier_error":null}}
```

### Configuration changes

PAPERLESS_OCR_LANGUAGE=deu, PAPERLESS_URL=<REDACTED>, USERMAP_UID=<REDACTED>, USERMAP_GID=<REDACTED>

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

**Comment by trurli** [comment]
Seems like there is no `Content-Disposition` header. 

RFC 2183 states:

> Content-Disposition is an optional header field. In its absence, the MUA may use whatever presentation method it deems suitable.

Would it be sensible then to handle attachments with no content disposition as content disposition "attachment" in `mail.py`? I might be able to create a pull request for this.

**Comment by stumpylog** [comment]
The logic in mail.py doesn't support what you are saying.  An empty content disposition would only matter for a rule which is attachments only.

**Comment by trurli** [comment]
Yes, of course, you are absolutely right. 

Apparently I wasn't quite awake early today and operated under the assumption that the system reimports unread emails, although it is explicitly stated in the dialog that only processed emails are marked as read. So with the first run all attachments were ignored because the rule was set to "attachments only" which I changed to "including inline" later. This of course never took effect.

I removed the list of processed emails directly from the database table and behold - the next import worked like a charm. Sorry for taking up your time.
